rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -205,6 +205,51 @@ rmw_connextdds_parse_string_list(
   return RMW_RET_OK;
 }
 
+rmw_ret_t rmw_connextdds_extend_initial_peer_list(
+  const rmw_peer_address_t * const static_peers,
+  const size_t static_peer_count,
+  struct DDS_StringSeq * const out)
+{
+  if (static_peer_count == 0) {
+    return RMW_RET_OK;
+  }
+
+  if (static_peers == nullptr) {
+    RMW_CONNEXT_LOG_ERROR_A_SET(
+      "received nullptr static_peers but a static_peer_count of %lu",
+      static_peer_count);
+    return RMW_RET_ERROR;
+  }
+
+  const auto initial_length = DDS_StringSeq_get_length(out);
+  const auto new_seq_length = initial_length + static_peer_count;
+  DDS_StringSeq_ensure_length(out, new_seq_length, new_seq_length);
+  for (size_t s=0; s < static_peer_count; ++s) {
+    const auto index = initial_length + s;
+    const char * peer = static_peers[s].peer_address;
+    char ** const element_ref = DDS_StringSeq_get_reference(out, index);
+    RMW_CONNEXT_ASSERT(nullptr != element_ref);
+    if (nullptr != *element_ref) {
+      /* If some strings are still hanging around this dead space in the seq,
+         then free their memory. */
+      DDS_String_free(*element_ref);
+    }
+    const auto peer_char_len = strlen(peer);
+    *element_ref = DDS_String_alloc(peer_char_len);
+    if (nullptr == *element_ref) {
+      RMW_CONNEXT_LOG_ERROR_SET("failed to allocate string");
+      return RMW_RET_ERROR;
+    }
+
+    memcpy(*element_ref, peer, peer_char_len);
```

Comment:
I always get nervous when I see a combination of strlen and memcpy, since it can be easy to accidentally not copy the null terminator. I think that might actually be the case here, but I'm not 100% sure.

https://stackoverflow.com/questions/8262406/get-strlen-after-memcpy-in-c

Reply:
I tried to address this in 9aef6d1, please check my work.